server/tserver/src/main/java/org/apache/accumulo/tserver/Tablet.java

```diff
@@ -1761,33 +1763,41 @@ Scanner createScanner(Range range, int num, Set<Column> columns, Authorizations
     private ScanDataSource isolatedDataSource;
     private boolean sawException = false;
     private boolean scanClosed = false;
+    private Semaphore scannerSemaphore;
```

Comment:
A javadoc comment describing why a semaphore is used would be nice when looking at this code in the future.